pkg/3scale/amp/template/adapters/system.go

```diff
@@ -7,14 +7,15 @@ import (
 )
 
 type System struct {
+	FileStorageType component.SystemFileStorageType
 }
 
-func NewSystemAdapter() Adapter {
-	return NewAppenderAdapter(&System{})
+func NewSystemAdapter(fileStorageType component.SystemFileStorageType) Adapter {
```

Comment:
To allow configurability of the storage type at System level via the templates we've added a new attribute to the System template-specific struct